test/e2e/kube-ovn/pod/statefulset.go

```diff
@@ -0,0 +1,61 @@
+package pod
+
+import (
+	clientset "k8s.io/client-go/kubernetes"
+
+	"github.com/onsi/ginkgo/v2"
+
+	"github.com/kubeovn/kube-ovn/pkg/util"
+	"github.com/kubeovn/kube-ovn/test/e2e/framework"
+)
+
+var _ = framework.Describe("[group:pod]", func() {
+	f := framework.NewDefaultFramework("pod")
+
+	var cs clientset.Interface
+	var podClient *framework.PodClient
+	var stsClient *framework.StatefulSetClient
+	var stsName string
+	var image string
+
+	ginkgo.BeforeEach(func() {
+		cs = f.ClientSet
+		podClient = f.PodClient()
+		stsClient = f.StatefulSetClient()
+		stsName = "sts-" + framework.RandomSuffix()
+		if image == "" {
+			image = framework.GetKubeOvnImage(cs)
+		}
+	})
+	ginkgo.AfterEach(func() {
+		ginkgo.By("Deleting sts " + stsName)
+		stsClient.DeleteSync(stsName)
+	})
+
+	framework.ConformanceIt("Should support statefulset scale Replica", func() {
+		// add this case for pr https://github.com/kubeovn/kube-ovn/pull/3777
+		replicas := 3
+		labels := map[string]string{"app": stsName}
+
+		ginkgo.By("Creating statefulset " + stsName)
+		sts := framework.MakeStatefulSet(stsName, stsName, int32(replicas), labels, framework.PauseImage)
+		sts = stsClient.CreateSync(sts)
+		ginkgo.By("Delete pod for statefulset " + stsName)
+		pod2Name := stsName + "-2"
+		pod2 := podClient.GetPod(pod2Name)
+		pod2IP := pod2.Annotations[util.IPAddressAnnotation]
+		err := podClient.Delete(pod2Name)
+		framework.ExpectNoError(err, "failed to delete pod "+pod2Name)
+		stsClient.WaitForRunningAndReady(sts)
+		pod2 = podClient.GetPod(pod2Name)
+		framework.ExpectEqual(pod2.Annotations[util.IPAddressAnnotation], pod2IP)
```

Comment:
This case has been inclued in `test/e2e/kube-ovn/ipam/ipam.go`, and there is no need to test it twice.

Reply:
fixed

Reply:
this is the necessary step for repeat bug